### DISCUSSION TITLE
Node Time Dimension ( Time Series )

### DISCUSSION THREAD

**Discussion by emdax** [ArcadeData/arcadedb#702] [Ideas]
It would be super intuitive and fun if Nodes ( and Edges ) could simply have a Time Dimension. Ideally it would be auto-magically activated per node when a MERGE or UPDATE is called on the node. This would enable every node to have Time Series functionality,  or FastForward and Rewind Time feature per node.

**Comment by lvca** [comment]
We could have a base Vertex/Edge/Document type such as:

```
type Timeable {
  long createdOn;
  long updatedOn;
}
```

So if your type wants to have the Time dimension could just extend Timeable and we could do a lot of things with that. In this case, the 2 long attributes are timestamp in ms.

**Comment by lvca** [comment]
Look also this proposal: https://github.com/ArcadeData/arcadedb/discussions/646

**Reply by emdax** [reply]
Yes definitely, I love that proposal as well. Versioning with timestamps would be marvelous.

**Comment by lvca** [comment]
We're close to starting the official development of the Time Series module after a lot of thoughts and requirements form users. You can followup here: https://github.com/ArcadeData/arcadedb/discussions/1180.
